orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionsSupport.java

```diff
@@ -0,0 +1,342 @@
+    context.registerFunction(name, ExpressionsSupport.class.getDeclaredMethod(name, types));
+  }
+
+  /**
+   * Creates a configured Spel evaluation context
+   * @param rootObject the root object to transform
+   * @param allowUnknownKeys flag to control what helper functions are available
+   * @return an evaluation context hooked with helper functions and correct ACL via whitelisting
+   */
+
+  static StandardEvaluationContext newEvaluationContext(Object rootObject, boolean allowUnknownKeys) {
+    StandardEvaluationContext evaluationContext = new StandardEvaluationContext(rootObject);
+    evaluationContext.setTypeLocator(new WhitelistTypeLocator());
+    evaluationContext.setMethodResolvers(Collections.singletonList(new FilteredMethodResolver()));
+    evaluationContext.setPropertyAccessors(Arrays.asList(new MapPropertyAccessor(allowUnknownKeys), new FilteredPropertyAccessor()));
+
+    try {
+      for (Map.Entry<String, List<Class<?>>> m : registeredHelperFunctions.entrySet()) {
+        registerFunction(evaluationContext, m.getKey(), m.getValue().toArray(new Class<?>[m.getValue().size()]));
+      }
+
+      if (allowUnknownKeys) {
+        // lazily function registering
+        registerFunction(evaluationContext, "fromUrl", String.class);
+        registerFunction(evaluationContext, "jsonFromUrl", String.class);
+        registerFunction(evaluationContext, "propertiesFromUrl", String.class);
+        registerFunction(evaluationContext, "stage", Object.class, String.class);
+        registerFunction(evaluationContext, "judgment", Object.class, String.class);
+        registerFunction(evaluationContext, "judgement", Object.class, String.class);
+      }
+    } catch (NoSuchMethodException e) {
+      // Indicates a function was not properly registered. This should not happen. Please fix the faulty function
+      LOGGER.error("Failed to register helper functions for rootObject {}", rootObject, e);
+    }
+
+    return evaluationContext;
+  }
+
+  /*
+    HELPER FUNCTIONS: These functions are explicitly registered with each invocation
+    To add a new helper function, append the function below and update ExpressionHelperFunctions and registeredHelperFunctions
+   */
+
+  /**
+   * Parses a string to an integer
+   * @param str represents an int
+   * @return an integer
+   */
+
+  public static Integer toInt(String str) {
+    return Integer.valueOf(str);
+  }
+
+  /**
+   * Parses a string to a float
+   * @param str represents an float
+   * @return an float
+   */
+
+  public static Float toFloat(String str) {
+    return Float.valueOf(str);
+  }
+
+  /**
+   * Parses a string to a boolean
+   * @param str represents an boolean
+   * @return a boolean
+   */
+
+  public static Boolean toBoolean(String str) {
+    return Boolean.valueOf(str);
+  }
+
+  /**
+   * Encodes a string to base64
+   * @param text plain string
+   * @return converted string
+   */
+
```

Comment:
Primarily from a consistency perspective: It's very atypical; certainly not something we do anywhere else in our projects.